### ISSUE TITLE
CAS-Management app won't reload new services

### ISSUE THREAD

**Issue by nkovacne** [Jasig/cas#1288]
We're using CAS 4.1.1. We have 2 nodes and both are using JDBC as backend for services, using the same DB (MySQL). Both nodes have the cas-management webapp deployed, so any of them might be used to manage services but we want them to be common on both nodes.

When we add a service on node A, node B's management interface won't reload that service until cas-management redeployment. Even waiting for the "Hibernate: select ... from RegisteredServiceImpl abstractre0_" event wouldn't refresh the service list. We've made some additional testing and services are indeed being reloaded correctly as far as CAS functionality goes, it's only the listing that gets not refreshed.

Our managementConfigContext.xml is:

```
    <bean
        id="dataSource"
        class="com.mchange.v2.c3p0.ComboPooledDataSource"
        p:driverClass="${database.driverClass:org.hsqldb.jdbcDriver}"
        p:jdbcUrl="${database.url:jdbc:hsqldb:mem:cas-ticket-registry}"
        p:user="${database.user:sa}"
        p:password="${database.password:}"
        p:initialPoolSize="${database.pool.minSize:6}"
        p:minPoolSize="${database.pool.minSize:6}"
        p:maxPoolSize="${database.pool.maxSize:18}"
        p:maxIdleTimeExcessConnections="${database.pool.maxIdleTime:1000}"
        p:checkoutTimeout="${database.pool.maxWait:2000}"
        p:acquireIncrement="${database.pool.acquireIncrement:16}"
        p:acquireRetryAttempts="${database.pool.acquireRetryAttempts:5}"
        p:acquireRetryDelay="${database.pool.acquireRetryDelay:2000}"
        p:idleConnectionTestPeriod="${database.pool.idleConnectionTestPeriod:30}"
        p:preferredTestQuery="${database.pool.connectionHealthQuery:select 1}"
    />

    <bean class="org.springframework.orm.jpa.support.PersistenceAnnotationBeanPostProcessor"/>

    <util:list id="packagesToScan">
        <value>org.jasig.cas.services</value>
    </util:list>

    <bean class="org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter"
          id="jpaVendorAdapter"
          p:generateDdl="true"
          p:showSql="true" />

    <bean id="entityManagerFactory"
          class="org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean"
          p:dataSource-ref="dataSource"
          p:jpaVendorAdapter-ref="jpaVendorAdapter"
          p:packagesToScan-ref="packagesToScan">
        <property name="jpaProperties">
            <props>
                <prop key="hibernate.dialect">${database.dialect:org.hibernate.dialect.HSQLDialect}</prop>
                <prop key="hibernate.hbm2ddl.auto">update</prop>
                <prop key="hibernate.jdbc.batch_size">${database.batchSize:1}</prop>
            </props>
        </property>
    </bean>

    <bean id="transactionManager" class="org.springframework.orm.jpa.JpaTransactionManager"
          p:entityManagerFactory-ref="entityManagerFactory" />

    <tx:advice id="txAdvice" transaction-manager="transactionManager">
        <tx:attributes>
            <tx:method name="delete*" read-only="false"/>
            <tx:method name="save*" read-only="false"/>
            <tx:method name="update*" read-only="false"/>
            <tx:method name="get*" read-only="true"/>
            <tx:method name="*" />
        </tx:attributes>
    </tx:advice>

    <aop:config>
        <aop:pointcut id="servicesManagerOperations" expression="execution(* org.jasig.cas.services.JpaServiceRegistryDaoImpl.*(..))"/>
        <aop:advisor advice-ref="txAdvice" pointcut-ref="servicesManagerOperations"/>
    </aop:config>

    <bean id="serviceRegistryDao"
          class="org.jasig.cas.services.JpaServiceRegistryDaoImpl" />
```

Properties are:

```
database.hibernate.dialect=org.hibernate.dialect.MySQLDialect
database.dialect=org.hibernate.dialect.MySQLDialect
database.batchSize=50
database.driverClass=com.mysql.jdbc.Driver
database.url=jdbc:mysql://fqdn:3306/cas?autoReconnect=true
database.user=***
database.password=***
```

**Comment by mmoayyed** [comment]
ok, thanks for the report. Will review this, but this likely wont find its way into 4.1.2.

**Comment by mmoayyed** [comment]
If you can step into the code and debug the issue, that would be most helpful.

**Comment by nkovacne** [comment]
I'm not familiar with the CAS Management code, but I guess I can give it a try when I set aside some time. I'll get back with the results if I find anything useful.

**Comment by jeremywickham** [comment]
I have noticed this behavior when using LDAP backed-ended services.

**Comment by mmoayyed** [comment]
is it a single ldap backend or do you also have 2 notes? and updates to the one are not recognized by the other?

**Comment by jeremywickham** [comment]
A single LDAP.

**Comment by mmoayyed** [comment]
OK. that makes it easy to troubleshoot then.

**Comment by mmoayyed** [comment]
I was able to recreate the issue. This will be available in 4.1.3
